src/App.vue

```diff
@@ -718,23 +713,37 @@ export default class App extends Mixins(
 
     // fetch draft filing
     // NB: will throw if API error
-    let draftFiling = await this.fetchDraftIA()
+    let draftFiling = await this.fetchDraftApplication()
+
+    // Protect the routes not associated with this filing type.
+    if (this.$route.meta.filingType !== this.getFilingType) {
+      console.log('ROUTE ERROR') // eslint-disable-line no-console
+      this.invalidRouteDialog = true
+    }
 
     // check if filing is in a valid state to be edited
-    this.invalidIncorporationApplicationDialog = !this.hasValidFilingState(draftFiling)
-    if (this.invalidIncorporationApplicationDialog) return
+    if (!this.hasValidFilingState(draftFiling)) return
 
     // merge draft properties into empty filing so all properties are initialized
-    const emptyFiling = this.buildIncorporationFiling()
-    draftFiling = { ...emptyFiling, ...draftFiling }
-
-    // parse draft filing into the store
-    if (draftFiling) {
-      this.parseIncorporationsDraft(draftFiling)
+    let resources, parseFiling: Function
```

Comment:
Doesn't this also declare `resources` as a Function? If so, that's not right.

Reply:
I'm not sure, i didn't think so but i can look closer at this. Will do some testing.

Reply:
Confirmed that this syntax doesn't cast the initial definition as well.

Reply:
OK, I just learned something!

How about putting those on separate lines?

Reply:
Absolutely! 
Updated in next commit